### ISSUE TITLE
Update user credentials

### ISSUE THREAD

**Issue by kbuffardi** [ChicoState/ResponSurv#6]
Log in is no longer required. Instead, upon first visiting ResponSurv, create a user account that is identified by the IP address (and timestamped). 

The approach can be similar to [Lazy Registration](https://www.bignerdranch.com/blog/lazy-user-registration-for-rails-apps/) but the user will _never_ actually register

**Comment by kbuffardi** [comment]
Upon user creation, also record:
- Mobile or Desktop
- Browser window size
- Max screen size
- Browser/agent

**Comment by cqcallaw** [comment]
I'm a bit concerned about the IP address and timestamp being sufficiently unique; what if a participant leaves the computer, and their IP address is dynamically re-assigned? Wouldn't they have to restart the survey?

How do you feel about generating a [UUID](https://en.wikipedia.org/wiki/Universally_unique_identifier) and storing it on the participant's computer in a cookie? This seems to be the basic approach used in the linked article about Lazy Registration

**Comment by kbuffardi** [comment]
Yeah, you're right, that's the idea. Although the user isn't aware of creating an account, we make a unique ID for them the first time they visit. We do want to record the IP and timestamp along with that.

**Comment by cqcallaw** [comment]
@kbuffardi Do you see any reason for the survey session to persist for longer than a month? I ask because I'm configuring the expiration dates of the cookies we'll be using to persist the session.

**Comment by cqcallaw** [comment]
@kbuffardi Another question: do we care that the browser window size might change if the participant resizes the window while they're taking the survey?

**Comment by kbuffardi** [comment]
A month should definitely be long enough.

And no, just the initial browser window size should be good enough -- ultimately, I'd be surprised if they actually need this data but I wanted to collect it "just in case"

**Comment by cqcallaw** [comment]
Okely dokely. The mechanism for logging the user agent is implemented; the information that requires client-side queries or survey questions (e.g. mobile vs. desktop) will take a little more work.
